src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -26,114 +28,239 @@
             }
-            i++;
 
-            if (pet != null) {
-                ArrayList<PetType> ts = (ArrayList<PetType>) clinicService.findPetTypes();
-                for (int j = 0; j < ts.size(); j++) {
-                    if (ts.get(j).getName().toLowerCase().equals(field)) {
-                        pet.setType(ts.get(j));
+            Pet pet = createPet(name, date, type);
+
+            // Beware of setting owner before deletion. This will result in no deletion occurring.
+            if(commandType.equals("add")){
+                pet.setOwner(owner);
+                clinicService.savePet(pet);
+            }else{ // already validated command types, and we only have two commands. This should be the delete command.
+                List<Pet> pets = owner.getPets();
+
+                for(Pet pet1: pets){
+                    if(pet1.equals(pet)){
+                        clinicService.deletePet(pet1);
                         break;
                     }
                 }
             }
 
-            field = "";
-            while (i < csv.length() && (csv.charAt(i) != ';' && csv.charAt(i) != '\n')) {
-                field += csv.charAt(i++);
-            }
+            pet.setOwner(owner); // setting owner here only for the http response
+            response.add(pet);
+        }
 
-            if (pet != null) {
-                String owner = field;
-                List<Owner> matchingOwners = clinicService.findAllOwners()
-                    .stream()
-                    .filter(o -> o.getLastName().equals(owner))
-                    .collect(Collectors.toList());
-
-                if (matchingOwners.size() == 0) {
-                    HttpHeaders headers = new HttpHeaders();
-                    headers.add("errors", "Owner not found");
-                    return new ResponseEntity<List<Pet>>(headers, HttpStatus.BAD_REQUEST);
-                }
-                if (matchingOwners.size() > 1) {
-                    HttpHeaders headers = new HttpHeaders();
-                    headers.add("errors", "Owner not unique");
-                    return new ResponseEntity<List<Pet>>(headers, HttpStatus.BAD_REQUEST);
-                }
-                pet.setOwner(matchingOwners.iterator().next());
-            }
+        return new ResponseEntity<>(response, HttpStatus.OK);
+    }
 
-            if (csv.charAt(i) == ';') {
-                i++;
+    private String[] parseCommand(String command){
+        return command.split(COMMAND_DELIMITER);
+    }
 
-                field = "";
-                while (i < csv.length() && csv.charAt(i) != '\n') {
-                    field += csv.charAt(i++);
-                }
+    private Date parseDate(String dateString){
+        Date date = new Date();
+        try {
+            date = new SimpleDateFormat("yyyy-MM-dd").parse(dateString);
+        }catch (ParseException e){
+            // already handled in validateDate, should never reach this clause
+        }
```

Comment:
Then this is redundant

Reply:
No, Java won't allow you to no handle the exception (use try-catch), otherwise it won't even compile.